internal/app/networkd/pkg/nic/nic.go

```diff
@@ -414,3 +406,89 @@ func (n *NetworkInterface) Reset() {
 	// nolint: errcheck
 	n.rtnlConn.LinkDown(link)
 }
+
+// addRoute is a function loosely copied from https://github.com/jsimonetti/rtnetlink/blob/154ecd417600f79d7847278a1e984056dc647acc/rtnl/route.go
+// and merged with our logic on determining gw, src, dst, etc.
+// we need this b/c we need to craft the route message ourselves to add attributes.
+// nolint: gocyclo
```

Comment:
We should contribute the changes back upstream so that we can get any other other goodies that may be fixed/changed in the future.

Reply:
Yep, let's chat in the AM and I'll run you through the current blockers and what we might upstream.